flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/rank/UpdatableTopNFunction.java

```diff
@@ -75,6 +77,13 @@
     private final InternalTypeInfo<RowData> rowKeyType;
     private final long cacheSize;
 
+    // flag to skip records with non-exist error instead to fail, true by default.
+    private final boolean lenient = true;
+
+    // data converter for logging only.
+    private final DataStructureConverter rowConverter;
```

Comment:
This will break compatibility. We can just introduce two transient fields.

Reply:
Good catch! transient is better.